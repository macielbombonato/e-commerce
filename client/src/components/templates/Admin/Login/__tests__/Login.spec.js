import createTestContext from 'react-cosmos-test/enzyme';

import fixture from '../Login.fixture';

const {mount, getWrapper, get} = createTestContext({fixture});

beforeEach(mount);

describe('#Login template', () => {
  it('renders component', () => {
    expect(getWrapper()).toMatchSnapshot();
  });

  it('logs the user in and send it to product listing page', () => {
    getWrapper('.btn-success').simulate('click');

    expect(get('props').history.push).toBeCalledWith('/admin/product');
  });
});
