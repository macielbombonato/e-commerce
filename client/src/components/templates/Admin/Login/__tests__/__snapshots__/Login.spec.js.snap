// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`#Login template renders component 1`] = `
<Login
  history={
    Object {
      "push": [Function],
    }
  }
>
  <Container
    tag="div"
  >
    <div
      className="container"
    >
      <Form
        tag="form"
      >
        <form
          className=""
        >
          <h1
            className="display-4"
          >
            Login
          </h1>
          <FormGroup
            tag="div"
          >
            <div
              className="form-group"
            >
              <Label
                for="username"
                tag="label"
                widths={
                  Array [
                    "xs",
                    "sm",
                    "md",
                    "lg",
                    "xl",
                  ]
                }
              >
                <label
                  className=" form-control-label"
                  htmlFor="username"
                >
                  Username
                </label>
              </Label>
              <Input
                defaultValue="admin"
                id="username"
                name="username"
                placeholder="John DOe"
                tag="p"
                type="text"
              >
                <input
                  className="form-control"
                  defaultValue="admin"
                  id="username"
                  name="username"
                  placeholder="John DOe"
                  type="text"
                />
              </Input>
            </div>
          </FormGroup>
          <FormGroup
            tag="div"
          >
            <div
              className="form-group"
            >
              <Label
                for="password"
                tag="label"
                widths={
                  Array [
                    "xs",
                    "sm",
                    "md",
                    "lg",
                    "xl",
                  ]
                }
              >
                <label
                  className=" form-control-label"
                  htmlFor="password"
                >
                  Password
                </label>
              </Label>
              <Input
                defaultValue="123"
                id="password"
                name="password"
                tag="p"
                type="password"
              >
                <input
                  className="form-control"
                  defaultValue="123"
                  id="password"
                  name="password"
                  type="password"
                />
              </Input>
            </div>
          </FormGroup>
          <Button
            color="success"
            onClick={[Function]}
            tag="button"
          >
            <button
              className="btn btn-success"
              onClick={[Function]}
              type="button"
            >
              Submit
            </button>
          </Button>
        </form>
      </Form>
    </div>
  </Container>
</Login>
`;
